lib/App/MintTag.pm

```diff
@@ -493,6 +495,41 @@ sub _maybe_delete_source_branches ($self, $step) {
   }
 }
 
+sub _maybe_cleanup_tags ($self, $step) {
+  return unless $step->push_tag_to && $step->cleanup_tag_days;
+
+  my $remote = $step->push_tag_to;
+  my $prefix = $step->tag_prefix;
+
+  try {
+    my $tag_spec = "refs/tags/${prefix}*";
+    my $tag_list = run_git('for-each-ref', "--format=%(refname:short)\t%(creatordate:unix)", $tag_spec);
+
+    my @tags = map {; my ($tag, $ts) = split /\t/; { ts => $ts, 'tag' => $tag} }
+    split /\n/, $tag_list;
+
+    my $time = time;
+
+    my $max_ts = $time - ($step->cleanup_tag_days * 86400);
+
+    my @can_remove = map {; $_->{tag} } grep {; $_->{ts} < $max_ts } @tags;
+
+    while (my @batch = splice @can_remove, 0, 100) {
+      $Logger->log_debug(["Deleting tags: %s", join(' ', @batch)]);
```

Comment:
I think we should use `$Logger->log(...)` here instead of log_debug, as this kind of info would be good to have in log records

Reply:
I still think we should use log here. "Hey why did tag foo disappear?!" "Oh.. our process of creating tags also destroys tags, forgot about that"

Reply:
I added the extra log statement in line 532 for that.

Reply:
That just says the number of entries we deleted, not which ones. I would record them personally, up to you.